### PR TITLE
[ChannelSelection.py] Code fixes and improvements

### PR DESCRIPTION
**Pull request by IanSav**
- Fix the code that sets the screen titles.  This includes adding some new and appropriate titles to some of the bouquet selection screens.
- Clean up and fix the ChannelSelection context menu screen.
- Improve the HistoryZapSelector screen.
- Improve the BouquetSelector screen.

### CODE REVIEW

File: po/ar.po [hunk 1]
```diff
@@ -6594,10 +6594,10 @@ msgstr ""
 msgid "Djibouti"
 msgstr "جيبوتي"
 
-msgid "Do center DVB subs on this service"
+msgid "Center DVB Subs On This Service"
 msgstr ""
 
-msgid "Do not center DVB subs on this service"
+msgid "Don't Center DVB Subs On This Service"
```
Comment (wedebe) [inline po/ar.po:6600]: (This and all others):
Menu items shouldn't be capitalised, only headings and brand names - eg.
`Don't Center DVB Subs On This Service` -> `Don't center DVB subs on this service`

English doesn't follow the German-style capitalisation of nouns.

**Review by wedebe — CHANGES_REQUESTED** [review]
(This and all others):
Menu items shouldn't be capitalised, only headings and brand names - eg.
Don't Center DVB Subs On This Service -> Don't center DVB subs on this service

(further more, insignificant words aren't title-cased in strings - Seven of Nine, This is Another, Comedy and Drama, Amsterdam to Zurich, Food for Thought, Wedebe is on a Rant, etc...)

The English language doesn't follow the German-style capitalisation of nouns.

**Comment by wedebe** [comment]
Not sure whether it's how I copied the files to my box, but I can't exit the channel list menu.

The non-numbered entries show an empty, mis-shapen bubble. Previously there was no icon.

**Comment by wedebe** [comment]
Attempting to change bouquet from the channel list triggers a crash

```
02:09:38.6495   File "/usr/lib/enigma2/python/Screens/ChannelSelection.py", line 2249, in channelSelected
02:09:38.6533     if ParentalControl.parentalControl.isServicePlayable(ref, self.bouquetParentalControlCallback, self.session):
02:09:38.6541        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
02:09:38.6545 AttributeError: type object 'ParentalControl' has no attribute 'parentalControl'
02:09:38.6547 [ePyObject] (PyObject_CallObject(<bound method ActionMap.action of <Components.ActionMap.ActionMap object at 0xaf3037f0>>,('OkCancelActions', 'ok')) failed)


```

File: po/ar.po (hunk 1 above)
Comment (IanSav) [reply po/ar.po:6600]: Menu entries are being capitalised.  (Titles, menus and button labels are being displayed capitalised. Descriptions, help text, etc are normal sentence style.)

**Comment by IanSav** [comment]
> Not sure whether it's how I copied the files to my box, but I can't exit the channel list menu.

Please explain how to reproduce this issue.

> The non-numbered entries show an empty, mis-shapen bubble. Previously there was no icon.

The previous code did not implement the list correctly.  This is a skin issue.  These "mis-shapen bubble" things are meant to look like the number icons but have no digit.  In my skin the icon contains a dot.

Sample

> Attempting to select any menu item results in an (X) icon appearing top-left of screen.

There was a keymap update missing from the commit.  I will add it now.

Apart from all of this, have I fixed *all* the ChannelSelection screen title issues?

**Comment by IanSav** [comment]
> Attempting to change bouquet from the channel list triggers a crash
> 
> ```
> 02:09:38.6495   File "/usr/lib/enigma2/python/Screens/ChannelSelection.py", line 2249, in channelSelected
> 02:09:38.6533     if ParentalControl.parentalControl.isServicePlayable(ref, self.bouquetParentalControlCallback, self.session):
> 02:09:38.6541        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> 02:09:38.6545 AttributeError: type object 'ParentalControl' has no attribute 'parentalControl'
> 02:09:38.6547 [ePyObject] (PyObject_CallObject(<bound method ActionMap.action of <Components.ActionMap.ActionMap object at 0xaf3037f0>>,('OkCancelActions', 'ok')) failed)
> ```

Can I please have complete instructions on how to reproduce this crash?

**Comment by wedebe** [comment]
> > Attempting to change bouquet from the channel list triggers a crash
> > ```
> > 02:09:38.6495   File "/usr/lib/enigma2/python/Screens/ChannelSelection.py", line 2249, in channelSelected
> > 02:09:38.6533     if ParentalControl.parentalControl.isServicePlayable(ref, self.bouquetParentalControlCallback, self.session):
> > 02:09:38.6541        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> > 02:09:38.6545 AttributeError: type object 'ParentalControl' has no attribute 'parentalControl'
> > 02:09:38.6547 [ePyObject] (PyObject_CallObject(<bound method ActionMap.action of <Components.ActionMap.ActionMap object at 0xaf3037f0>>,('OkCancelActions', 'ok')) failed)
> > ```
> 
> Can I please have complete instructions on how to reproduce this crash?
> 
> By the way, please remember that this topic was about fixing the titles in ChannelSelection. You have not commented on this issue. We are hoping to close your issue report if this issue is fixed. While we welcome reporting of other issues please remember to respond to the core issue.

I can't respond to the core issue if I can't select a different bouquet, which you've now fixed, so I'll be able to test again later today ;)

**Comment by wedebe** [comment]
To reproduce the crash, I just open the channel list, select `Favorites` to open the bouquet list and choose any of the bouquet names. I've never configured parental control, but having set it up, that doesn't appear to solve the problem.

**Comment by wedebe** [comment]
Btw, perhaps a whitespace would fill the dot? To me, a dot-within-a-dot looks unnecessarily busy/cluttered.

**Comment by jbleyel** [comment]
> Btw, perhaps a whitespace character would fill the dot? To me, a dot-within-a-dot looks unnecessarily busy/cluttered.

This issue is fixed in ChoiceList.py -> https://github.com/openatv/enigma2/commit/6cb6b5b04d9ef286f312bbf5763ab67f286114a4

**Comment by jbleyel** [comment]
> To reproduce the crash, I just open the channel list, select `Favorites` to open the bouquet list and choose any of the bouquet names. I've never configured parental control, but having set it up, that doesn't appear to solve the problem.
> 
> (openATV 2023-05-19 mutant51 with changed files copied)

The crash should be fixed with the changes of ParentalControl.

**Comment by jbleyel** [comment]
Hi @wedebe , can you please make a final test?
Thanks

**Comment by wedebe** [comment]
Something in https://github.com/openatv/enigma2/pull/2894/files is causing a bootloop for me now. I won't have time to look into this until later in the week.

**Comment by IanSav** [comment]
What is the crash?
Logs?

**Pull request merged**